docs(profiling): Add page about performance overhead (#6739)
getsentry/sentry-docs: src/docs/product/profiling/performance-overhead.mdx

---
title: Performance Overhead
sidebar_order: 4
redirect_from:
  - /profiling/performance-overhead/
description: "How enabling profiling impacts the performance of your application."
---

> Profiling works by taking snapshots (samples) of the call stack of each thread in your application at a regular interval, and transmitting that data to event ingestion servers over the public internet. This method of profiling is known as sampling profiling, and all of our profilers collect samples approximately 100 times per second (or once every 10ms). Sampling profiling provides a reasonable tradeoff between data fidelity and runtime overhead, as opposed to deterministic profiling, which introduces significantly higher overhead by tracking every function call.
> 

Reviewer: ```suggestion
Profiling works by taking snapshots (samples) of the call stack of each thread in your application at a regular interval, then transmitting that data to event ingestion servers via the Internet. This method of profiling is known as sampling profiling. Sentry's profilers collect samples approximately 100 times per second (or once every 10ms). Sampling profiling provides a reasonable tradeoff between data fidelity and runtime overhead, as opposed to deterministic profiling, which introduces significantly higher overhead by tracking every function call.

```

> **For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.**
> 

Reviewer: ```suggestion
<Note>
For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.
</Note>

```

> The quantitative overhead of profiling (measured as a % increase in CPU time) is highly dependent on many environmental factors such as the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. However, as a general rule of thumb, we target an approximate **5-10% CPU time overhead** on a typical configuration with mid-end hardware on each platform.

Reviewer: ```suggestion
The quantitative overhead of profiling, (measured as a % increase in CPU time), is highly dependent on many environmental factors such as: the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. However, as a general rule of thumb, Sentry targets an approximate **5-10% CPU time overhead** on a typical configuration with mid-end hardware on each platform.
```

---
title: Performance Overhead
sidebar_order: 4
redirect_from:
  - /profiling/performance-overhead/
description: "How enabling profiling impacts the performance of your application."
---

Profiling works by taking snapshots (samples) of the call stack of each thread in your application at a regular interval, then transmitting that data to event ingestion servers via the Internet. This method of profiling is known as sampling profiling. Sentry's profilers collect samples approximately 100 times per second (or once every 10ms). Sampling profiling provides a reasonable tradeoff between data fidelity and runtime overhead, as opposed to deterministic profiling, which introduces significantly higher overhead by tracking every function call.

> <Note>
> For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.
> </Note>

Reviewer 2: ```suggestion
<Note>

For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.

</Note>
```

Reviewer 2: I remember reading that we need these extra newlines for weird Gatsby reasons

> The quantitative overhead of profiling (measured as a % increase in CPU time) is highly dependent on many environmental factors such as the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. However, as a general rule of thumb, we target an approximate **1-5% CPU time overhead** on a typical configuration with mid-end hardware on each platform.

Reviewer 2: ```suggestion
The quantitative overhead of profiling (measured as a percentage increase in CPU time) is dependent on environmental factors such as the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. As a rule of thumb, we target approximately **1-5% CPU time overhead** on a typical configuration with mid-end hardware on each platform.
```

Reviewer 2: this felt wordy, so I simplified a bit
